### ISSUE TITLE
Enhance notification response

### ISSUE THREAD

**Issue by adrianq** [EyeSeeTea/FIRE-MockServer#17]
@MurhafSousli  to propose a new response format

**Comment by adrianq** [comment]
@MurhafSousli  can you make a proposal for the new format? @tokland it is about to start working with these minor changes on the web servers side and he would like to take care of all of them at one

**Comment by MurhafSousli** [comment]
@adrianq  @tokland This is the current interface for the notification
```ts
export interface Notification {
  id: number;
  created: Date;
  seen: boolean;
  type: string;
  newUserRequest: UserRequest;
  newUserAccepted: UserRequest;
  message: Message;
  voucher: Voucher;
  user: User;
}

export interface UserRequest {
  id: number;
  state: string;
  adminUser: User;
  user: User;
  created: Date;
  updated: Date;
}

export interface Message {
  id: number;
  text: string;
  fromUser: User;
  toUser: User;
  created: Date;
}

export interface Voucher {
  id: number;
  state: string;
  url: string;
  Vendor: string;
  bulkNumber: string;
  code: string;
  creditRemaining: number;
  creditTotal: number;
  depleted: any;
  user: User;
  activated: Date;
  created: Date;
}
```
This is the desired format
```ts
export interface Notification {
  id: number;
  created: Date;
  seen: boolean;
  type: string;
  user: User;
  content: UserRequest | Message | Voucher | null;
}
```
related #15

**Comment by tokland** [comment]
> content: UserRequest | Message | Voucher | null;

I am not sure about this, we would be losing information regarding that field, it's too implicit IMO. Also, should we need more than one field for some kind of notification, where would we store it? What do you think of using ADT/discriminated union tags?

https://blog.mariusschulz.com/2016/11/03/typescript-2-0-tagged-union-types

```
type Notification = newUserRequest | newUserAccepted | ...
```

I guess you are wary of too-verbose switchs/ifs chain, but the nature of this kind of data-structure forces you to check what you have before using. You can build functions to extract components from a Notification, so it's kept abstracted.

By the way, what does this "user" field in Notification represent? it's that's not in the ws models, I'll have to add it.

**Comment by tokland** [comment]
You removed this but I think it's worth commenting it:

> I also want to mention that we are handling the http errors using the catch operator, so you might want to flatten and remove the `"status": "success"

I followed flask recommendations but I am with you, I am not sure we need this ok/error when we have HTTP responses codes. I'd say we can flatten single objects, but leave it for collections: {data: [], ...} or more explicit {messages: [], ...}. When we add pagination we'll need to return a JSON object with the pager.

Let's keep that in mind when we implement the real ws.

**Comment by MurhafSousli** [comment]
@adrianq @tokland Another change suggestion!

In the notification list we display the type of the notification as in the screenshot.

![image](https://user-images.githubusercontent.com/8130692/29694409-a9f588c6-8944-11e7-99c2-9991ee2a413e.png)

It would be better if we can translate the type directly using our current translation scheme

```
"NOTIF": {
      "TOPPED_UP": "Topped up",
      "USER_REQUEST": "New User request",
      "USER_ACCEPTED": "New User accepted",
      "MESSAGE_SENT": "Message sent",
      "PROFILE_UPDATED": "Profile updated"
}
```
for instance instead of `profileUpdated` return `PROFILE_UPDATED`

This way we will be able to bind it directly like
```
{{ notification.type | translate }}
```
otherwise we should change the translation scheme to match type's value.

**Comment by MurhafSousli** [comment]
@tokland
> I guess you are wary of too-verbose switchs/ifs chain

That's right, we already have tons of checks like the translation pipe and others.. If we got overloaded with checks, we might have performance issues on the old phones especially that it is a hybrid app. 
I think checking the `type` of the notification is sufficient.

> You can build functions to extract components from a Notification

I would avoid using functions to modify the response as possible as I can. Rendering the data directly is more beautiful, our code will be less and more focused IMHO.

> By the way, what does this "user" field in Notification represent?

The current server returns the  `user` field when the notification is `profileUpdated`

![image](https://user-images.githubusercontent.com/8130692/29694826-bc5fde96-8947-11e7-92fd-50668f3797b8.png)
